src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -198,10 +135,10 @@ public boolean shutdown() throws InterruptedException {
     int terminateTimeout = (shutdownTimeout - gracefulTimeout) - 2;
 
     logger.info("JobService request Stairway shutdown");
-    boolean finishedShutdown = stairway.quietDown(gracefulTimeout, TimeUnit.SECONDS);
+    boolean finishedShutdown = stairwayComponent.get().quietDown(gracefulTimeout, TimeUnit.SECONDS);
```

Comment:
Another way to think about this is that this the same as keeping `stairway` as a field in this class that's initialized in the `initialize` method as the result of `stairwayComponent.get()`. If you wrote it that way, it would have the same semantics and you could revert all the changes where `stairway` was replaced with a call to `stairwayComponent.get()`.

Reply:
That's a good point!